src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/GnosisControlSafeDialog.tsx

```diff
@@ -63,25 +91,146 @@ type Props = DialogProps &
   Partial<WizardDialogType<object>> &
   ActionDialogProps;
 
-const validationSchema = (isPreview) =>
-  yup.object().shape({
+const GnosisControlSafeDialog = ({
+  colony,
+  cancel,
+  callStep,
+  prevStep,
+  isVotingExtensionEnabled,
+}: Props) => {
+  const [showPreview, setShowPreview] = useState(false);
+  const [selectedContractMethods, setSelectedContractMethods] = useState<{
+    [key: number]: AbiItemExtended | undefined;
+  }>();
+  const [expandedValidationSchema, setExpandedValidationSchema] = useState<
+    Record<string, any>
+  >({});
+
+  const getMethodInputValidation = useCallback(
+    (inputType: string, contractName: string, isArraySchema?: boolean) => {
+      if (inputType.slice(-2) === '[]') {
+        return yup.array().when('contractFunction', {
+          is: (contractFunction) => contractFunction === contractName,
+          then: yup
+            .array()
+            .ensure()
+            .of(
+              getMethodInputValidation(
+                inputType.slice(0, -2),
+                contractName,
+                true,
+              ),
+            ),
+          otherwise: false,
+        });
+      }
+      if (inputType === 'uint256' || inputType === 'int256') {
+        return yup.number().when('contractFunction', {
+          is: (contractFunction) =>
+            contractFunction === contractName || isArraySchema,
+          then: yup
+            .number()
+            .test(
+              'is-integer',
+              () => MSG.notIntegerError,
+              (value) => new Decimal(value || 0).isInteger(),
+            )
```

Comment:
Think you could use `number().integer()` here?

Reply:
I didn't use it because `.integer()` uses `Number.isInteger()` in the background, which may not be as precise as using Decimal.